### PR TITLE
Truncate support for field-level filtering

### PR DESCRIPTION
**Pull request by trek-td**
re: https://tranquildata.atlassian.net/browse/HC-24

Support for truncating structs down to a subset of fields

Note to reviewers: you can ignore most of the `models/tranquil_*.go` files as they're autogenerated. The only ones you need to examine are: `tranquil_explanationofbenefit.go`, `tranquil_tuple_support.go`, and `tranquil_tuple_support_test.go`

### CODE REVIEW

**Review by jiri-td — COMMENTED** [review]
Other than the go.mod charades (yikes), the 12k+ likes of changes are in syntax and adjustments, but no new conceptual work....

**Comment by trek-td** [comment]
Yes, sorry that I forgot to annotate this PR. The changes to the `tranquil_<resourcetype>.go` files are generated code for the `Truncate` method. The only one of any significance is the one in `tranquil_explanationofbenefit.go`. I'll add a comment to that file.

File: models/tranquil_explanationofbenefit.go [hunk 1]
```diff
@@ -1,169 +1,223 @@
...
+ 		"related": &FieldTypeSupport{"ExplanationOfBenefitRelatedClaimComponent", true, false},
+ 		"prescription": &FieldTypeSupport{"Reference", false, true},
+ 		"diagnosis": &FieldTypeSupport{"ExplanationOfBenefitDiagnosisComponent", true, false},
+ 		"precedence": &FieldTypeSupport{"uint32", false, true},
+ 
+	 }
+ }
+ 
+ func (fhirVal *ExplanationOfBenefit) Truncate(fieldToKeep string) (TupleSupport, error) {
+	 if strings.ToLower(fieldToKeep) == "benefitbalance" {
```
Comment (trek-td) [inline models/tranquil_explanationofbenefit.go:178]: This is the one 'real' implementation of `Truncate`

**Review by john-td — APPROVED** [review]
The truncate function in ExplanationOfBenefit and the gomod changes look good.

**Pull request merged**